### DISCUSSION TITLE
Link to open embedded images

### DISCUSSION THREAD

**Discussion by Ericbaz** [py-pdf/fpdf2#310] [General]
Hey, everyone. I've been building a complex document over the last couple of months and they contain some big mineral scan images that need to be able to be clicked so the client can view them on their original size, which is usually huge.

I'm aware of set_link() and the link parameter on the image() method, but neither really help me as they're either for document navigation or opening external files.

The PDF document my work is based on was made on Acrobat Pro, so the scientists could create a link that open the resized embedded image itself, nothing from the outside.

I was wrecking my brain today but couldn't really get the links to function like that. Is there such a functionality on this library?

Sorry if this isn't the right place to ask this. Really appreciate your work on this, by the way. It helped me a ton already!

**Comment by Lucas-C** [comment]
> a link that open the resized embedded image itself, nothing from the outside

I fear that the PDF file format does not provide such functionality...

What you'd like is for readers to be able to click on a "miniature" image in the PDF document,
and the OS-program associated with this file type would launch: is that correct?
The same way an OS user can usually double-click in an explorer window on an image file and it gets opened with a viewer (`PhotoViewer.dll` on Windows, [eog](https://wiki.gnome.org/Apps/EyeOfGnome) under Ubuntu...)

In fact the PDF spec defines some custom link actions, as annotations: [GoToAction / GoToRemoteAction / LaunchAction](https://github.com/PyFPDF/fpdf2/blob/master/test/test_annotations.py#L51)
However from my experience, nowadays, those actions are not usually implemented by PDF readers, for security reasons.

**Comment by gmischler** [comment]
> The PDF document my work is based on was made on Acrobat Pro, so the scientists could create a link that open the resized embedded image itself, nothing from the outside.

You make it sound like this is something that already exists in other tools.
If so, could you prepare an example with Acrobat Pro to show how it works?
The smallest possible file that demonstrates the functionality would be ideal. An analysis of that file would then reveal whether it makes sense to implmenent something similar for fpdf.
